src/Common/src/Common/Extensions/ExceptionExtensions.cs

```diff
@@ -59,4 +59,21 @@ public static bool IsCancellation(this Exception? exception)
 
         return false;
     }
+
+    /// <summary>
+    /// Determines whether the thrown exception results from an HTTP request timeout.
+    /// </summary>
+    /// <param name="exception">
+    /// The caught exception to inspect.
+    /// </param>
+    public static bool IsTimeout(this Exception? exception)
```

Comment:
This doesn't look right; I would expect such a method to also return `true` when the incoming exception is a `TimeoutException`. Either rename to `IsHttpClientTimeout`, or adapt the code so it becomes generally usable.

Should validate other places in the codebase where `TimeoutException` appears.

Reply:
Yeah, that thought crossed my mind, but I didn't spend much time on it since it's all internal... Do you have a preference for making this generic vs specific?

Reply:
I'm fine making it specific, we don't have other places in the codebase where it matters.